File: 02/2.py
#!/bin/python3
from functools import reduce


def line_power(sets: list[str]) -> int:
    minimums = {"red": 0, "green": 0, "blue": 0}
    for set in sets:
        for subset in [x.strip() for x in set.split(",")]:
            [amount, color] = subset.split(" ")
            minimums[color] = max(int(amount), minimums[color])
    return reduce(lambda x, acc: acc * x, minimums.values(), 1)


with open("input.txt") as f:
    print(
        sum(
            line_power(line[1].split(";"))
            for line in [x.split(":") for x in f.readlines()]
        )
    )

File: 06/2.exs
#!/bin/elixir

# {:ok, text} = File.read("test.txt")
{:ok, text} = File.read("input.txt")

[t | [d]] = text |> String.split("\n") |> Enum.map(fn l -> l |> String.split(~r"\W+") |> tl |> Enum.reduce("", fn str, acc -> "#{acc}#{str}" end) |> String.to_integer end) |> IO.inspect()

for i <- 1..(t-1), reduce: 0 do x ->
  cond  do
    i*(t-i) > d -> x+1
    true -> x
  end
end |> IO.inspect()